### DISCUSSION TITLE
Changelog

### DISCUSSION THREAD

**Discussion by jasonwilliams** [community/community#128442] [npm]
One of the main reasons I visit the NPM website is because there's a major (breaking) change, and I need to understand what has changed so I know if I can upgrade or not.

NPM doesn't show any of this data, so the next step is the github repo, which is an extra hop.
Would it be possible to parse out the changelog and show on NPM? Or at least point directly to a changelog.

## Prior Art
Visual Studio Code market place have a changelog link on [their page](https://marketplace.visualstudio.com/items?itemName=pflannery.vscode-versionlens) for each extension which pulls down the changelog.md file from the repository. It sounds like a small thing but being able to see the changelog directly in VSCode saves me a minute of navigating to the repo to look at it there.

Of course some people prefer to use the release section to write release notes instead so it could be used as a fallback, im sure you could use the Github API to find the release (using the same version on NPM) and returning data from there, and if there's nothing using the changelog.md

**Comment by MylesBorins** [comment]
@jasonwilliams this is a great idea. I really like the idea of bringing the release on GitHub closer to the package published on npm and making it easier to navigate between the two. I'm going to mull over a few ideas of how we can improve this and get back to you.

**Comment by mheuzey** [comment]
I would like to second this, and also add that sometimes the GitHub repository might be set to private for some reason, so a user might not always be able to head over there to view the changelog.

**Comment by dartess** [comment]
What about adding a new field to the package.json: "changelog"? The changelog can be in different formats, it can be a release page in the github, it can be a separate MD file or a page on the official website. The site can take this field and display it in the sidebar.
@MylesBorins Any thoughts?

**Comment by kisaragi-hiu** [comment]
Currently the process to find the changelog is unnecessarily convoluted. One has to

1. go to each dependency's npm page
2. guess if the package keeps its changelog in its website or in its repository
3. if it's in the website: visit the website and look for a changelog
4. if it's in the repository, guess if the changelog is kept in
    - CHANGELOG.md
    - (for monorepos) packages/<package>/CHANGELOG.md
    - (as) GitHub Releases
    - or somewhere else

just to find the changelog, or even whether there is one.

###### Proposal

Add a new package.json field, `changelog`, similar to `homepage` or `bugs`, for a URL pointing to a package's Changelog. This allows package maintainers to declare where they keep their changelog.

The NPM website can then utilize this field, just like `homepage`, to show a link to the Changelog if it has been declared.

This will also encourage people to keep a changelog, as it is more likely to be read.

<figure>
<img
     height="300"
     src="https://user-images.githubusercontent.com/11722318/232024754-810b20e3-add5-4512-97be-fb76848b1f5f.png"
     alt="An NPM package metadata section with a Changelog field">
<figcaption><em>What a Changelog field can look like. It doesn't matter too much if the link is ellipsized - the user would just click on it.</em></figcaption>
</figure>
